File: python/7kyu/Digit*Digit.py
def square_digits(num):
    if num == 0:
        return 0
    
    num_arr = list(map(int, str(num)))
    result = []
    for el in num_arr:
        result.append(el ** 2)
    
    return int("".join([str(n) for n in result]))

File: typescript/7kyu/Digit*Digit.ts
export class Kata {
  static squareDigits(num: number): number {
    if (num === 0) return 0;
    let result = num
      .toString()
      .split("")
      .map((el) => {
        return parseInt(el) * parseInt(el);
      })
      .join("");
    return Number(result);
  }
}
